### ISSUE TITLE
Detecting mpep stimulus times

### ISSUE THREAD

**Issue by sylviaschroeder** [cortex-lab/kilotrodeRig#1]
Hi @nsteinme ,
just went over some old recordings where I presented high frequency flickers using mpep. The correct time of stimulus on- and offset is really crucial in these recordings. Now I noticed 2 lines in the alignment script that I don't understand (line 233 in alignAll.m):
```
tpd = Timeline.rawDAQData(:,strcmp({Timeline.hw.inputs.name}, 'photoDiode'));
sig = conv(diff([0; tpd]).^2, ones(1,16/1000*Fs), 'same');
```

Basically, the photodiode signal is differentiated, squared and then convolved with a square pulse.
Because of the squaring, however, one can't differentiate between up and down flips in the photodiode signal anymore, and the convolution spreads out the time of each flip. I was wondering why you coded it like this.

Thanks!

**Comment by nsteinme** [comment]
Yeah I think that was the point - the idea is to make a signal from which you can easily detect the first frame of the mpep stimulus. Note that "flipsUp" and "flipsDown" here are NOT the times of every frame, just the times of the start of a stimulus and the end of it. E.g. if you showed 5 stimuli for 100ms each, there are 5 starts/ends (aka flipsUp/flipsDown) even though the photodiode actually flips 6 times per stimulus for 30 total times. This code does not report the times of each frame. Knowing the starts/ends is the best place to start when looking for that, though - now you can detect up/downs of the raw photodiode between the starts/ends.

**Comment by sylviaschroeder** [comment]
Ok. I understand.
Thanks!
